install_files/ansible-base/roles/tails-config/files/securedrop_init.py

```diff
@@ -89,3 +99,16 @@
                  'SecureDrop successfully auto-configured!',
                  'You can now access the Journalist Interface.\n',
                  'If you are an admin, you can now SSH to the servers.'])
+
+# As the amnesia user, check for SecureDrop workstation updates.
+os.setresgid(amnesia_gid, amnesia_gid, -1)
+os.setresuid(amnesia_uid, amnesia_uid, -1)
+output = subprocess.check_output([path_securedrop_admin_venv,
+                                  path_securedrop_admin_init,
+                                  '--root', path_securedrop_root,
+                                  'check_for_updates'], env=env)
+
+lock_location = "/home/amnesia/Persistent/securedrop/securedrop_update.lock"
+if 'Update needed' in output or os.path.exists(lock_location):
```

Comment:
Can you please tell us about how that race condition can arrive?

Reply:
Double clicking it twice too fast? Not sure, but if we're trying to use a lock file, we might as well do it right.

Reply:
I looked into this some more, and it looks like we're also using that `.lock` as some sort of semaphore between two update-check and update-apply type commands. If we're not using the file to lock a resource, it might be better to name it `.flag`. We still *do* want to `flock` a `.lock` file somwhere when we start the update process. This would `try/except OSError` (I think that's what it throws) and say "another process is already updating" when it errors.

Reply:
>Double clicking it twice too fast? Not sure, but if we're trying to use a lock file, we might as well do it right.

The updater tool will fireup automatically after boot. There is no manual way to start the application for the users.

Reply:
This still feels like the right way to do it, and it would make it more robust against future changes. What you're doing looks like a soft lock, and this becomes a lock at the kernel level. If we don't actually need a lock, then we might as well get rid of this, no?

Reply:
Renamed in 0ebc321992e224d1973e18d71ee2c40fcc89ee8b